kernel/kernel_drivers.go

```diff
@@ -165,48 +187,145 @@ func createModulesSubtree(origin, dest string) error {
 		}
 	}
 
-	return nil
+	// If necessary, add modules from components and run depmod
+	return setupModsFromComp(dest, kversion, compInfos)
 }
 
-func driversTreeDir(ksnapName string, rev snap.Revision) string {
-	return filepath.Join(dirs.SnapdStateDir(dirs.GlobalRootDir), "kernel", ksnapName, rev.String())
+func driversTreeDir(kernelSubdir string, rev snap.Revision) string {
+	return filepath.Join(dirs.SnapdStateDir(dirs.GlobalRootDir), "kernel", kernelSubdir, rev.String())
 }
 
-// RemoveKernelDriversTree cleans-up the writable kernel tree for a given snap
-// and revision.
-func RemoveKernelDriversTree(ksnapName string, rev snap.Revision) (err error) {
-	treeRoot := driversTreeDir(ksnapName, rev)
+// RemoveKernelDriversTree cleans-up the writable kernel tree in snapd data
+// folder, under kernelSubdir/<rev> (kernelSubdir is usually the snap name).
+func RemoveKernelDriversTree(kernelSubdir string, rev snap.Revision) (err error) {
+	treeRoot := driversTreeDir(kernelSubdir, rev)
 	return os.RemoveAll(treeRoot)
 }
 
-// EnsureKernelDriversTree creates a tree in
-// /var/lib/snapd/kernel/<ksnapName>/<rev> with subfolders can be bind-mounted
-// on boot to /usr/lib/{modules,firmware}. This tree contains files from the
-// kernel snap mounted on kernelMount, as well as symlinks to it.
-// TODO this will be extended to consider kernel-modules components.
-func EnsureKernelDriversTree(ksnapName string, rev snap.Revision, kernelMount string) (err error) {
+// KernelModulesMntInfo contains information about a component and where it is
+// mounted.
+type KernelModulesMntInfo struct {
+	CompSideInfo *snap.ComponentSideInfo
+	MntDir       string
```

Comment:
why do we need MntDir here?

Reply:
I have changed this so the path is assumed in `/run/mnt/kernel-snaps/...` and this is not needed.